src/main/java/ch/njol/skript/events/SimpleEvents.java

```diff
@@ -685,6 +686,20 @@ public class SimpleEvents {
 				.requiredPlugins("MC 1.16+");
 		}
 
+		if (Skript.classExists("io.papermc.paper.event.player.PlayerDeepSleepEvent")) {
+			Skript.registerEvent("Player Deep Sleep", SimpleEvent.class, PlayerDeepSleepEvent.class, "player deep sleep[ing]")
+				.description(
+					"Called when a player has slept long enough to count as passing the night/storm.",
+					"Cancelling this event will prevent the player from being counted as deeply sleeping unless they exit and re-enter the bed."
+				)
+				.examples(
+					"on player deep sleeping:",
+					"\tsend \"Zzzz..\" to player"
+				)
+				.since("INSERT VERSION")
+				.requiredPlugins("Paper 1.16+");
```

Comment:
```suggestion
					.description(
						"Called when a player has slept long enough to count as passing the night/storm.",
						"Cancelling this event will prevent the player from being counted as deeply sleeping unless they exit and re-enter the bed."
					)
					.examples(
						"on player deep sleeping:",
							"\tsend \"Zzzz..\" to player"
					)
					.since("INSERT VERSION")
					.requiredPlugins("Paper 1.16+");
```